dxr/app.py

```diff
@@ -114,27 +122,38 @@ def _search_json(query, tree, query_text, is_case_sensitive, offset, limit, conf
             if is_case_sensitive:
                 params['case'] = 'true'
             return jsonify({'redirect': url_for('.browse', _anchor=line, **params)})
+
     try:
-        count_and_results = query.results(offset, limit)
-        # Convert to dicts for ease of manipulation in JS:
-        results = [{'icon': icon,
-                    'path': path,
-                    'lines': [{'line_number': nb, 'line': l} for nb, l in lines],
-                    'is_binary': is_binary}
-                   for icon, path, lines, is_binary in count_and_results['results']]
+        # Pull up all the non-negated text terms.
+        single_term = query.single_term()
+        count, results = query.results(offset, limit)
+        promoted_count, promoted, promoted_query = 0, [], ""
+        if offset == 0 and single_term:
+            # Now we pull out the single line term and pass into promoted paths.
+            try:
+                promoted_count, promoted, promoted_querystring = query.promoted_paths(single_term)
```

Comment:
This reads much better now. Maybe we should rename single_term() to single_text_term() to reflect its behavior change.